CRM/Core/BAO/Domain.php

```diff
@@ -69,7 +69,7 @@ public static function retrieve(&$params, &$defaults) {
    *
    * @return CRM_Core_BAO_Domain|null
    */
-  public static function &getDomain($reset = NULL) {
+  public static function getDomain($reset = NULL) {
```

Comment:
Saying this out loud to ensure we agree: changing this function signature is safe because:

 * There shouldn't be any subclasses (let alone folks overriding the function), and 
 * The function returns an object which means that the `&` should have negligible impact behavior (for better or worse or mystery, both signatures allow the caller to mutate the domain object)